ts/Extensions/OverlappingDataLabels.ts

```diff
@@ -89,7 +89,42 @@ function chartHideOverlappingLabels(
             box2.x + box2.width <= box1.x ||
             box2.y >= box1.y + box1.height ||
             box2.y + box2.height <= box1.y
-        );
+        ),
+        pointIsInPolygon = (p: Vec, polygon: Vec[]): boolean => {
+            const len = polygon.length,
+                { x: checkpointX, y: checkpointY } = p;
+            let inside = false;
+
+            for (let i = 0, j = len - 1; i < len; j = i++) {
+                const x1 = ~~polygon[i].x;
+                const y1 = ~~polygon[i].y;
+                const x2 = ~~polygon[j].x;
+                const y2 = ~~polygon[j].y;
+
+                if (
+                    (y1 > checkpointY) !== (y2 > checkpointY) && (
+                        checkpointX < (x2 - x1) *
+                        (checkpointY - y1) / (y2 - y1) +
+                        x1
+                    )
+                ) {
+                    inside = !inside;
+                }
+            }
+            return inside;
+        },
+        boxCheck = (box1: BBoxObject, box2: BBoxObject): boolean => {
+            const box1Poly = box1.poly,
```

Comment:
A possible performance improvement here would be to first check if the rectangular bounding boxes overlap. If they don't, we don't need to loop over all the vectors to check of an exact overlap.